tools/azure-sdk-tools/devtools_testutils/proxy_fixtures.py

```diff
@@ -16,15 +17,102 @@
 # the functions we patch
 from azure.core.pipeline.transport import RequestsTransport
 
-from .helpers import get_test_id, is_live_and_not_recording
+from .helpers import get_test_id, is_live, is_live_and_not_recording
 from .proxy_testcase import start_record_or_playback, stop_record_or_playback, transform_request
 from .proxy_startup import test_proxy
+from .sanitizers import add_general_regex_sanitizer
 
 if TYPE_CHECKING:
     from typing import Any, Callable, Dict, Optional, Tuple
     from pytest import FixtureRequest
 
 
+_LOGGER = logging.getLogger()
+
+
+class EnvironmentVariableSanitizer:
+    def __init__(self) -> None:
+        self._fake_values = {}
+
+    def sanitize(self, variable: str, value: str) -> str:
+        """Registers a sanitizer that replaces the value of the specified environment variable with the provided value.
+
+        :param str variable: Name of the environment variable to sanitize.
+        :param str value: Value to sanitize the environment variable's value with.
+
+        :returns: The real value of `variable` in live mode, or the sanitized value in playback.
+        """
+        self._fake_values[variable] = value
+        real_value = os.getenv(variable)
+        if real_value:
+            add_general_regex_sanitizer(regex=real_value, value=value)
+        else:
+            _LOGGER.info(f"No value for {variable} was found, so a sanitizer could not be registered for the variable.")
+
+        return real_value if is_live() else value
+
+    def sanitize_batch(self, variables: "Dict[str, str]") -> "Dict[str, str]":
```

Comment:
I am not going to block the PR on this because I think it'll actually work, but I'm curious.

[Our typehint docs](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/docstring_typehint.md#type-hints-for-python-35) show `Dict[str, str]` instead of `"Dict[str, str]"`. That's also the style that' I've been following in my code. What makes you choose the `surrounded by quotes`? There's gotta be a reason and I'd like to understand it!

Reply:
Oh I see. Because you have the conditional import above. Due to that, you HAVE to leave it string so it can resolve during type checking. Is that correct?

Reply:
Yep, that's right! To be honest, I think my conditional import is outdated -- I've seen libraries starting to import `typing` types regardless of `TYPE_CHECKING`. But if the import is conditional, you need the quotes to avoid runtime errors